packages/dds/ordered-collection/src/test/fuzzUtils.ts

```diff
@@ -0,0 +1,233 @@
+	validationStrategy: { type: "fixedInterval", interval: 10 },
+	clientJoinOptions: {
+		maxNumberOfClients: 6,
+		clientAddProbability: 0.05,
+	},
+	defaultTestCount: 100,
+	saveFailures: { directory: path.join(_dirname, "../../src/test/results") },
+};
+
+type FuzzTestState = DDSFuzzTestState<ConsensusQueueFactory>;
+
+export interface AddOperation {
+	type: "add";
+	value: string;
+}
+
+export interface AcquireOperation {
+	type: "acquire";
+}
+
+export interface ResolveOperation {
+	type: "resolve";
+	result: ConsensusResult;
+}
+
+/**
+ * Represents ConsensusOrderedCollection operation types for fuzz testing
+ */
+export type ConsensusOrderedCollectionOperation =
+	| AddOperation
+	| AcquireOperation
+	| ResolveOperation;
+
+function makeOperationGenerator(): Generator<
+	ConsensusOrderedCollectionOperation,
+	FuzzTestState
+> {
+	type OpSelectionState = FuzzTestState & {
+		itemValue: string;
+	};
+
+	const valuePoolRandom = makeRandom(0);
+	const dedupe = <T>(arr: T[]): T[] => [...new Set(arr)];
+	const valuePool = dedupe(
+		Array.from({ length: valueConfigs.valuePoolSize }, () =>
+			valuePoolRandom.string(valueConfigs.valueStringLength),
+		),
+	);
+
+	async function add(state: OpSelectionState): Promise<AddOperation> {
+		return {
+			type: "add",
+			value: state.itemValue,
+		};
+	}
+
+	async function acquire(state: OpSelectionState): Promise<AcquireOperation> {
+		return {
+			type: "acquire",
+		};
+	}
+
+	async function resolve(state: OpSelectionState): Promise<ResolveOperation> {
+		return {
+			type: "resolve",
+			result: state.random.pick([ConsensusResult.Complete, ConsensusResult.Release]),
+		};
+	}
+
+	const clientBaseOperationGenerator = createWeightedGenerator<
+		ConsensusOrderedCollectionOperation,
+		OpSelectionState
+	>([
+		[add, 1],
+		[acquire, 1],
+		[resolve, 1],
+	]);
+
+	return async (state: FuzzTestState) =>
+		clientBaseOperationGenerator({
+			...state,
+			itemValue: state.random.pick(valuePool),
+		});
+}
+
+function makeReducer(): Reducer<ConsensusOrderedCollectionOperation, FuzzTestState> {
+	const pendingCallbacks = new Map<string, string>();
+	const callbackResolver = new EventEmitter();
+
+	const reducer = combineReducers<ConsensusOrderedCollectionOperation, FuzzTestState>({
+		add: ({ client }, { value }) => {
+			client.channel.add(value).catch((error) => {
+				throw error;
+			});
+		},
+		acquire: ({ client }) => {
+			const callback = async (value: string): Promise<ConsensusResult> => {
+				const callbackId = uuid();
+				pendingCallbacks.set(callbackId, value);
+				return new Promise<ConsensusResult>((resolve) => {
+					const onCallbackResolve = (id: string, result: ConsensusResult): void => {
+						if (id === callbackId) {
+							callbackResolver.off("resolve", onCallbackResolve);
+							resolve(result);
+						}
+					};
+					callbackResolver.on("resolve", onCallbackResolve);
+				});
+			};
+			client.channel.acquire(callback).catch((error) => {
+				throw error;
+			});
```

Comment:
catch+throw here doesn't really help (same as just floating the promise).  I'm not sure if the fuzz tests have any handling for uncaught exceptions or unhandled rejections.

Is there another way to report this error to the fuzz test harness?